File: 2DVirtualWorldSimulatorCpp/2DVirtualWorldSimulator/Antelope.cpp
#include <iostream>
#include "Antelope.h"
#include "World.h"

Antelope::Antelope() {
	this->name = 'A';
	this->strength = 4;
	this->initiative = 4;
}

void Antelope::action(World* world) {
	//Has wider range of movement - 2 fields instead of 1.
	world->moveOrganism(this, TWO_STEPS);
}

bool Antelope::collision(World* world,Organism* organismToFight) {
	if (this->isAttacker == true) {
		this->defaultAttack(world, organismToFight);
		return true;
	}
	else {
		pos newPosition;
		int my_chances = rand() % 2;
		if (my_chances == 0) {
			newPosition = world->posFreeAround(this);
			if (newPosition.x != world->GetHorizSize() + 1) {
				world->setOrganismVectorData(nullptr, position.y, position.x);
				this->SetOrganismPosition(newPosition.x, newPosition.y);
				world->setOrganismVectorData(this, position.y, position.x);
				world->addCommentatorVector(MOVED, this, nullptr, NULL);
				return true;
			}
			else false;
		}
		else return false;
	}
}

char Antelope::GetName() const {
	return 'A';
}

string Antelope::GetOrganismFullName() const {
	return "Antelope";
}

Antelope::~Antelope() {
}

File: 2D_sym_python/animals/Antelope.py
from Animal import*
import random

class Antelope(Animal):
    def __init__(self):
        super().__init__()
        self.name='A'
        self.strength=4
        self.initiative=4
    
    def action(self, world):
        world.moveOrganism(self,TWO_STEPS)
    
    def collision(self, world, organismToFight):
        if self.isAttacker == True:
            self.defaultAttack(world,organismToFight)
            return True
        else:
            my_chances=random.randint(0,1)
            if my_chances == 0:
                newPosition = world.posFreeAround(self)
                if newPosition['x'] != world.GetHorizSize()+1:
                    world.SetOrganismVectorData(None, self.position['y'], self.position['x'])
                    self.SetOrganismPosition(newPosition['x'], newPosition['y'])
                    world.SetOrganismVectorData(self, self.position['y'], self.position['x'])
                    world.AddCommentatorVector(reportType.MOVED, self, None, 0)
                    return True
                else: return False
            else: return False

    def GetName(self):
        return 'A'
    
    def GetOrganismFullName(self):
        return "Antelope"